app/services/core/persistent-stateful-service.ts

```diff
@@ -38,4 +45,15 @@ export abstract class PersistentStatefulService<TState extends object> extends S
       },
     );
   }
+
+  runMigrations(persistedState: any, migrations: IMigration[]) {
+    const migratedState = cloneDeep(persistedState);
+    migrations.forEach(migration => {
+      if (!persistedState[migration.oldKey]) return;
+      migratedState[migration.newKey] = migration.transform(persistedState[migration.oldKey]);
+      migratedState[migration.oldKey] = null;
```

Comment:
`delete migratedState[migration.oldKey]`
This will actually remove the old key from the object instead of keeping and nulling it.